updater.go

```diff
@@ -176,8 +185,18 @@ func downloadFile(out *os.File, url, checksum string) error {
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
-	defer out.Close()
+	defer func() {
+		err = resp.Body.Close()
+	}()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = out.Close()
+	}()
+	if err != nil {
+		return err
+	}
```

Comment:
Same problem as mentioned above. How about moving the function calls to `Close()` to the end and return the error to not make it too complicated with `defer`?